### ISSUE TITLE
Evaluate m5.1 Mendelian VEP across inference context sizes

### ISSUE THREAD

**Issue by gonzalobenegas** [Open-Athena/marin-dna#485]
# Evaluate m5.1 Mendelian VEP across inference context sizes

## TL;DR

The six-context experiment is complete for one fixed 255 bp-trained m5.1 checkpoint on the Mendelian development cohort.
Cropping inference below 255 bp reduced both zero-shot and newly trained frozen-probe macro AUPRC.
Extending to 511 bp produced a small zero-shot gain that was statistically significant in the paired analysis but no significant probe change, while extending to 1023 bp sharply degraded both protocols.
The extension arms do not compare checkpoints trained at longer contexts.

## Goal

How does MarinDNA m5.1's Mendelian variant-effect-prediction performance change when its inference window is shortened below the 255 bp context used in training?
Do the context-response curves differ across consequence subsets?
How do the curves differ between zero-shot likelihood-ratio scoring and frozen-embedding linear probing?

## Description

MarinDNA m5.1 is the 1B Qwen3-style model highlighted in the [August 2026 blog post](https://www.openathena.ai/blog/marin-dna/).
The checkpoint is registered as `mix-v0.9-p1B-i24-exp135-m5.1-step-59158` and uses 255 DNA bases plus a BOS token.
The blog reports strong Mendelian VEP performance at that native context and explains that 255 bp was chosen to model individual functional elements efficiently.

[#37](https://github.com/Open-Athena/marin-dna/issues/37) compared models pretrained with 256 bp and 512 bp contexts on the Promoter VEP subset and found no clear difference.
That experiment changed the training context and model run.
The proposed experiment holds the checkpoint fixed and crops only the context supplied during VEP.
No matched m5.1 inference-time context ablation is recorded in the current research question, evaluation documentation, or accepted experiment pages.
The production frozen-embedding probe protocol developed in [#314](https://github.com/Open-Athena/marin-dna/issues/314) and [#320](https://github.com/Open-Athena/marin-dna/issues/320) can reuse embeddings emitted by the same scoring passes.

## Experiment design

Use one checkpoint, tokenizer, reference genome, variant dataset, split, scoring protocol, and metric implementation for every arm.
Change only the number of DNA bases extracted around each SNV.

| DNA context | Total tokens with BOS | Model ID | Role |
|---:|---:|---|---|
| 255 bp | 256 | `mix-v0.9-p1B-i24-exp135-m5.1-step-59158` | Existing native-context baseline |
| 127 bp | 128 | `mix-v0.9-p1B-i24-exp135-m5.1-step-59158-ctx127` | New configuration alias |
| 63 bp | 64 | `mix-v0.9-p1B-i24-exp135-m5.1-step-59158-ctx63` | New configuration alias |
| 31 bp | 32 | `mix-v0.9-p1B-i24-exp135-m5.1-step-59158-ctx31` | New configuration alias |

All four DNA windows are odd and place the SNV at the same central position on the forward and reverse-complement strands.
The shorter arms use the same checkpoint source as the 255 bp entry:

```text
gs://marin-us-east5/checkpoints/dna-bolinas-mix-v0.9-p1B-i24-exp135-zoonomia-m5.1-bef41e/hf/step-59158
```

Evaluate only `mendelian_traits` from `bolinas-dna/evals_mendelian_traits` at pinned revision `4aed58e50c5dea0b878a665007af2ef9e5108e9f` on the development `train` split.
Do not access held-out test labels, predictions, or aggregate metrics.
Use the canonical Ensembl release 115 GRCh38 soft-masked primary assembly already configured in `evals_v2`.

### Zero-shot evaluation

Use the existing zero-shot score `minus_llr_avg`.
This score averages the forward and reverse-complement raw LLRs and orients the result so pathogenic alternatives should rank above matched common alternatives.
Compute AUPRC and cluster-bootstrap standard error with the existing 1,000-iteration, seed-0 `match_group` bootstrap.
Report every subset emitted by the pinned dataset and the `_macro_avg_` row.
The macro average remains the unweighted mean over subsets with at least 30 match groups, following the existing leaderboard contract.
Exclude `_global_` from the requested zero-shot figure.

### Frozen-embedding linear probe

Emit pooled reference and alternate embeddings for every arm during the same forward and reverse-complement scoring passes.
The pipeline mean-pools each DNA window with BOS excluded, averages the forward and reverse-complement embeddings in float32, and stores them in float16.

Run the existing per-consequence-subset frozen linear probe independently at every context size.
For Mendelian variants, use `concat_ref_delta`, the concatenation of the reference embedding and `alternate - reference` embedding.
Upcast features to float32 and use the production `StandardScaler` plus L2-regularized logistic-regression protocol with leave-one-chromosome-out predictions and inner chromosome-grouped tuning of `C`.
Keep the existing eligibility requirements and hyperparameter grid unchanged.

Use `probe_score` to report per-subset and `_macro_avg_` AUPRC with the existing per-chromosome weighting and 1,000-iteration, seed-0 chromosome-cluster bootstrap.
The probe metrics artifact also contains a paired `minus_llr_avg` baseline evaluated on the probe rows, but that diagnostic is not directly level-comparable to the standard matched-pair zero-shot metric and is not part of either requested figure.

## Pipeline plan

The current `evals_v2` scoring path accepts these windows through per-model `window_size` configuration.
`transform_llr_clm` extracts an arbitrary centered window, and the model runner derives the strand-aware variant position from that window.
The 31 bp arm still satisfies the score kernel's requirement for a shared prefix and at least one downstream prediction.

Register the three shorter arms as new model entries in `snakemake/analysis/evals_v2/config/config.yaml` with the shared `gcs_path`, their respective `window_size`, and `datasets: [mendelian_traits]`.
Add all four arms to `probe.models` for `mendelian_traits`.
Use unique model IDs so the new score and metric parquets cannot overwrite the native 255 bp artifacts.
Do not add the ablation arms to the public model leaderboard because they are inference configurations of one trained checkpoint rather than distinct models.

For each context, produce scores with `config/overlays/return_embeddings.yaml` so the score parquet contains `emb_ref` and `emb_alt` as well as the zero-shot scores.
Use the targeted `--forcerun compute_scores` procedure documented by `evals_v2` when an existing score artifact must be regenerated with embeddings.
Reuse the native 255 bp embedding artifact only after verifying its schema and provenance.

For example, extract embeddings for the 127 bp arm with:

```bash
uv run --locked --group genome-s3 snakemake \
  --configfile config/overlays/return_embeddings.yaml \
  --forcerun compute_scores -- \
  results/scores/mix-v0.9-p1B-i24-exp135-m5.1-step-59158-ctx127/mendelian_traits.parquet
```

Dry-run the four zero-shot metric targets, four probe artifacts, and four probe-metric targets before launching any work.
Representative targets for one shortened arm are:

```bash
uv run --locked --group genome-s3 snakemake -n \
  results/metrics/mix-v0.9-p1B-i24-exp135-m5.1-step-59158-ctx127/mendelian_traits.parquet \
  results/probe/mix-v0.9-p1B-i24-exp135-m5.1-step-59158-ctx127/mendelian_traits.parquet \
  results/probe_metrics/mix-v0.9-p1B-i24-exp135-m5.1-step-59158-ctx127/mendelian_traits.parquet \
  --rerun-triggers mtime
```

Inspect the dry-run for unintended models, datasets, or upstream work before launching remote compute.
Run the targeted embedding extraction and probe targets with `--rerun-triggers mtime` so the committed default `return_embeddings: false` does not cause Snakemake to rebuild score artifacts without embeddings.
Obtain approval before launching paid remote compute.
The existing checkpoint-cache path is keyed by model ID, so the three aliases may materialize the same checkpoint three times.
Record the duplicate storage and transfer cost; optimize checkpoint aliasing separately only if that cost is material.

## Figures

Create one faceted line plot for the standard zero-shot matched-pair AUPRC and a separate faceted line plot for the frozen-probe per-chromosome-weighted AUPRC.
Do not combine the two metrics on a shared y-axis because their estimators and eligible rows differ.

In each figure, use one panel for every eligible consequence subset and one panel for the macro average.
Use a shared logarithmic base-2 x-axis labeled `Inference context (bp)` with explicit ticks at 31, 63, 127, and 255.
Use a shared y-axis within each figure and label it with the corresponding AUPRC definition.
Show the four point estimates connected by straight line segments and add vertical error bars for ±1 cluster-bootstrap SE.
Do not smooth the four-point series.
Mark the 255 bp native-context point with a distinct shape, and distinguish the macro panel through both its title and line weight rather than color alone.
Label zero-shot panels whose subset does not enter the macro average because it has fewer than 30 match groups.
Show probe subsets that fail the probe eligibility requirements as missing and do not connect line segments across missing points.
Export both reviewed figures as SVG and retain the compact source tables used to generate them.

## Analysis boundaries

Treat comparisons based on the plotted standard errors as descriptive.
The four arms score the same matched variants, so a formal claim about pairwise differences requires a paired `match_group` bootstrap of the AUPRC delta.
The current pipeline computes one standard error per arm and does not compute that paired delta.
Formal probe deltas would analogously require a paired chromosome bootstrap.

The zero-shot LLR includes the variant-position term and a sum over downstream positions.
Shrinking the window changes both the available biological sequence and the number of downstream likelihood terms.
AUPRC removes sensitivity to a uniform score rescaling, but it does not remove changes in variant ranking caused by the shorter sum.

For the probe, shrinking the window changes both the sequence available to the model and the number of positions entering mean pooling.
Each context size gets its own probe trained on embeddings produced at that context.
The probe comparison therefore measures decodability under a fixed supervised protocol, not the robustness of one classifier transferred across context sizes.

This experiment measures inference-time cropping and extension for a model pretrained at 255 bp.
It cannot determine the best pretraining context, whether other checkpoints or tasks benefit from inference-only extension, or the best architecture for long-range dependencies.
The conclusions apply to SNVs in the Mendelian development cohort and the specified zero-shot and frozen-probe protocols.

## Relationship to the context-size research question

Broaden [#392](https://github.com/Open-Athena/marin-dna/issues/392) and `docs/research/questions/long-context.md` from long-context acquisition to context size across pretraining, inference, downstream adaptation, model scale, and biological task.
Use `Context size` as the root README label.
A precise revised question could be: "What context sizes do genomic language models need for different biological tasks, and how should models acquire and use that context?"
Retain the existing long-context strategy synthesis as one part of the broader question.
Relate this experiment as evidence about inference-time context sensitivity for one fixed short-context model.

## Completion criteria

- The three shorter context aliases are registered without changing the existing 255 bp artifact namespace.
- All four arms are registered for the Mendelian frozen-probe workflow.
- The targeted Snakemake dry-run contains only the expected checkpoint, score, zero-shot metric, probe, and probe-metric jobs for the four Mendelian cells.
- Every arm uses the pinned development split and produces finite `minus_llr_avg` scores for every row.
- Every score parquet used for probing contains valid `emb_ref` and `emb_alt` columns with the expected dimensions.
- Every arm produces a probe artifact and probe-metrics artifact under the unchanged production probe protocol.
- Within each evaluation protocol, per-subset and macro AUPRC rows use the expected sample counts and subset membership across all four arms, with any ineligible probe subset recorded explicitly.
- The existing 255 bp score and metric artifacts are reused only when their schema and provenance match this experiment.
- Two reviewed SVGs show every eligible dataset subset plus macro average at 31, 63, 127, and 255 bp with ±1 protocol-appropriate cluster-bootstrap SE.
- The issue conclusion reports the observed context-size relationships and states the paired-inference and pretraining-context limitations.
- The knowledge-base disposition is recorded as `pending`, `interpretation PR open`, `interpretation page merged`, `reusable lesson merged elsewhere`, or `no promotion because no scientifically valid claim remains`.

## Status

The experiment, paired analyses, artifact validation, and reviewed figures are complete.
All issue clusters are terminated, and estimated SkyPilot cost was approximately $3.09.
Independent and human interpretation review are complete.
PR #492 merged as [`015ff228`](https://github.com/Open-Athena/marin-dna/commit/015ff22827164ea7328df096d82913152078f695).

## Links

- Coordinating research question: [#392](https://github.com/Open-Athena/marin-dna/issues/392)
- Accepted question page: [Context size](https://github.com/Open-Athena/marin-dna/blob/015ff22827164ea7328df096d82913152078f695/docs/research/questions/long-context.md)
- Accepted experiment page: [Inference-context sensitivity of m5.1 Mendelian VEP](https://github.com/Open-Athena/marin-dna/blob/015ff22827164ea7328df096d82913152078f695/docs/research/experiments/485-m5-1-inference-context.md)
- Interpretation: [PR #492](https://github.com/Open-Athena/marin-dna/pull/492)
- General remote-bootstrap fix: [PR #493](https://github.com/Open-Athena/marin-dna/pull/493)
- Research snapshot: [commit `064532e`](https://github.com/Open-Athena/marin-dna/commit/064532e96eca3caf9e25a57f70b955e39ede4de0)
- Logbook: [commit-pinned record](https://github.com/Open-Athena/marin-dna/blob/064532e96eca3caf9e25a57f70b955e39ede4de0/.agents/logbooks/issue-485-m5-1-inference-context.md)
- W&B report: not used
- Important updates: [final result comment](https://github.com/Open-Athena/marin-dna/issues/485#issuecomment-5363664447)

## Decision Log

- Evaluate one fixed checkpoint at 255, 127, 63, and 31 bp inference contexts.
- Run the existing zero-shot and frozen-embedding probe protocols at every context size and report them separately.
- Record observations without directional predictions.
- Use only the Mendelian development split for scoring, probing, model selection, and interpretation.

## Conclusion

For the fixed m5.1 checkpoint, both evaluation protocols benefited from inference context up to the 255 bp training length.
The 511 bp extension improved zero-shot macro AUPRC by +0.0175 with paired p=0.0024 but did not significantly change the trained-probe macro result.
The 1023 bp extension reduced both macro metrics by about 0.30 relative to 511 bp, with paired p < 0.0002 at the two-sided 10,000-draw resolution.
This supports matching inference context to the checkpoint and task; it does not determine the best pretraining context or long-context architecture.

## Next steps

The accepted [Context size question](https://github.com/Open-Athena/marin-dna/blob/015ff22827164ea7328df096d82913152078f695/docs/research/questions/long-context.md#possible-directions) records matched training-context comparisons, fixed-probe transfer, long-range tasks, and context-acquisition experiments as possible directions.

## Knowledge-base disposition

`interpretation page merged`

**Comment by gonzalobenegas** [comment]
🤖 Final result: m5.1 inference-context sensitivity

- Change: completed zero-shot scoring and newly trained frozen probes for one fixed 255 bp-trained m5.1 checkpoint at 31, 63, 127, 255, 511, and 1023 bp on the pinned Mendelian development `train` split.
- Result: macro AUPRC rises from 0.1941 to 0.3945 zero-shot and from 0.3174 to 0.4779 with the probe between 31 and 255 bp.
- Paired inference: 255→511 bp changes zero-shot macro AUPRC by +0.0175 (95% CI [+0.0057, +0.0278], p=0.0024) and probe macro AUPRC by -0.0124 (95% CI [-0.0408, +0.0087], p=0.3138).
- Failure boundary: 511→1023 bp changes macro AUPRC by -0.3060 zero-shot and -0.2993 with the probe; both paired intervals exclude zero with p < 0.0002 at the two-sided 10,000-draw resolution.
- Scope: 511 and 1023 bp are inference-only extrapolations, not checkpoints trained at those contexts; no held-out labels or metrics were accessed.
- Reproducibility:
  - [Interpretation PR #492](https://github.com/Open-Athena/marin-dna/pull/492)
  - [Research snapshot](https://github.com/Open-Athena/marin-dna/commit/064532e96eca3caf9e25a57f70b955e39ede4de0)
  - [Zero-shot figure](https://github.com/Open-Athena/marin-dna/blob/3d15954dafdaff07b0bc3d083f4a5fd8ca23fa88/.agents/artifacts/issue-485-context-size/zero_shot_context.svg)
  - [Probe figure](https://github.com/Open-Athena/marin-dna/blob/3d15954dafdaff07b0bc3d083f4a5fd8ca23fa88/.agents/artifacts/issue-485-context-size/probe_context.svg)
  - [Logbook](https://github.com/Open-Athena/marin-dna/blob/064532e96eca3caf9e25a57f70b955e39ede4de0/.agents/logbooks/issue-485-m5-1-inference-context.md)
- Operations: all issue clusters are terminated; estimated SkyPilot cost was approximately $3.09.
- Knowledge-base disposition: `interpretation PR open`.
- Next: human interpretation review of #492 before merge.

**Comment by gonzalobenegas** [comment]
🤖 Review update: interpretation PR

- Independent review found two issues in #492: correct the two-sided 10,000-draw resolution from exact p=0.0001 to p < 0.0002, and update a stale inbound link after broadening the question to Context size.
- Fix: both were addressed in [`952b509`](https://github.com/Open-Athena/marin-dna/commit/952b509a7589dca2c6d653c5933e8d3007e8bd80).
- Re-review: no actionable findings.
- Validation: the full PR CI rerun is green.
- Next: human interpretation review before merge.

**Comment by gonzalobenegas** [comment]
🤖 General fix extracted

- [PR #493](https://github.com/Open-Athena/marin-dna/pull/493) separately pins the CPU probe Sky task to the project's exact `uv` version before its locked sync.
- The regression checks every locked Sky task against `tool.uv.required-version`, and the same PR corrects the embedding-overlay batch size in the README from 32 to 96.
- Validation: the focused test and targeted pre-commit checks pass.
- Scope: no issue #485 experiment aliases, results, figures, or interpretation are included.

**Comment by gonzalobenegas** [comment]
🤖 Experiment complete

- Accepted interpretation: [PR #492](https://github.com/Open-Athena/marin-dna/pull/492) merged as [`015ff228`](https://github.com/Open-Athena/marin-dna/commit/015ff22827164ea7328df096d82913152078f695).
- Result: from 31 to the 255 bp training context, macro AUPRC rose from 0.1941 to 0.3945 zero-shot and from 0.3174 to 0.4779 with newly trained frozen probes.
- Inference extension: 255→511 bp produced a small paired zero-shot gain (+0.0175, p=0.0024) and no significant probe change; 511→1023 bp reduced both macro metrics by about 0.30 (p < 0.0002 at the two-sided 10,000-draw resolution).
- Scope: the evidence covers one fixed 255 bp-trained checkpoint, one Mendelian development cohort, and two evaluation protocols; no held-out labels or metrics were accessed.
- Durable record: the accepted [experiment page](https://github.com/Open-Athena/marin-dna/blob/015ff22827164ea7328df096d82913152078f695/docs/research/experiments/485-m5-1-inference-context.md) records the findings and limitations, and the accepted [Context size question](https://github.com/Open-Athena/marin-dna/blob/015ff22827164ea7328df096d82913152078f695/docs/research/questions/long-context.md) records possible next experiments.
- General fix: [PR #493](https://github.com/Open-Athena/marin-dna/pull/493) merged separately for the remote `uv` bootstrap issue found during this experiment.
- Knowledge-base disposition: `interpretation page merged`.
